django-rgd/rgd/rest/get.py

```diff
@@ -44,3 +44,14 @@ class GetSpatialAsset(RetrieveAPIView, _PermissionMixin):
     serializer_class = serializers.SpatialAssetSerializer
     lookup_field = 'spatial_id'
     queryset = models.SpatialAsset.objects.all()
+
+
+# TODO: temporary until viewset refactor
+class GetUserCollections(ListAPIView, _PermissionMixin):
+    serializer_class = serializers.CollectionSerializer
+    queryset = models.Collection.objects.all()
+
+    def get_queryset(self):
+        if self.request.user.id is not None:
+            return super().get_queryset().filter(collection_permissions__user=self.request.user)
+        return None
```

Comment:
This needs to handle the collections that super users can see but might not be assigned. Further, if the setting `RGD_GLOBAL_READ_ACCESS` is enabled, then any `Collection` with no `CollectionPermission` should return as well.

@mvandenburgh, would you please implement a new method called `get_user_collections()` in `django-rgd/rgd/permissions.py` that will handle this logic then have the endpoint call that?

`get_user_collections()` should look *something* like:

```py
def get_user_collections(user):
    queryset = models.Collection.objects.all()
    # Called outside of view
    if user is None:
        return queryset  # Not sure about this case?
    # Must be logged in
    if not user.is_active or user.is_anonymous:
        return queryset.none()
    # Superusers can see all (not staff users)
    if user.is_superuser:
        return queryset
    # Else, check user permissions
    condition = Q(collection_permissions__user=self.request.user)
    if getattr(settings, 'RGD_GLOBAL_READ_ACCESS', False):
        condition |= Q(**{'collection_permissions__isnull': True})
    return queryset.filter(condition)
```

Reply:
Implemented this in https://github.com/ResonantGeoData/ResonantGeoData/pull/529/commits/898fb7a5c8e406b6dca467038e65731909d1a6d2 with a slight modification.